### PR TITLE
Improve Page Hits tracking performance

### PR DESCRIPTION
**Pull request by xnohat**
Improve Page Hits tracking performance by change other of code lines

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging for features or enhancements / 3.0 for bug fixes 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no

#### Description:
Discovery impact to server load performance of Page Hits tracking when server get heavy load traffic

#### Explain:
PageModel->hitPage() method will overload Server because it query LeadModel to get Lead very heavy double times, to fix that, change order of codes, get Lead first by getCurrentLead()  and pass as argument to hitPage(), getCurrentLead() is lightweight and fast execution, hitPage() will skip query current Lead again and using passed Lead.
Execution Time of page hits processing will change from 0.2 us down to 0.01 us, increase performance 2000% . PHP-FPM will not get full connections any time later

#### Steps to test this PR:
This PR require manual test
1. Load test by Apache Bench with 5000 concurrent connection in 5 minutes and get 1st server load report
2. Load up [this PR](https://mautibox.com)
3. Load test by Apache Bench with 5000 concurrent connection in 5 minutes and get 1st server load report
Server load immediately drop many times ( depend on server hardware specs )

### CODE REVIEW

**Comment by mabumusa1** [comment]
@xnohat if you need help getting this PR merged, please let me know. I think this is a very good code improvement

**Comment by RCheesley** [comment]
Thanks for your PR @xnohat and for the detailed instructions for testing.

Could you take a look at signing the contributors agreement on the link above please?

**Comment by xnohat** [comment]
> Thanks for your PR @xnohat and for the detailed instructions for testing.
> 
> Could you take a look at signing the contributors agreement on the link above please?

I have done for signing the contributors agreement

**Comment by RCheesley** [comment]
@cla-bot check

File: app/bundles/PageBundle/Controller/PublicController.php [hunk 1]
```diff
@@ -365,9 +365,15 @@ public function previewAction($id)
      */
     public function trackingImageAction()
     {
+         /** @var LeadModel $leadModel */
+        $leadModel = $this->getModel('lead');
+
+        $lead = $leadModel->getCurrentLead();
```
Comment (fedys) [inline app/bundles/PageBundle/Controller/PublicController.php:371]: Method `LeadModel::getCurrentLead()` no longer exists in the `staging` branch. You should use `\Mautic\LeadBundle\Tracker\ContactTracker::getContact()` instead.

File: app/bundles/PageBundle/Controller/PublicController.php [hunk 2]
```diff
@@ -365,9 +365,15 @@ public function previewAction($id)
...
+        
         /** @var \Mautic\PageBundle\Model\PageModel $model */
         $model = $this->getModel('page');
-        $model->hitPage(null, $this->request);
+        //$model->hitPage(null, $this->request);
+        $model->hitPage(null, $this->request, '200', $lead);
```
Comment (fedys) [inline app/bundles/PageBundle/Controller/PublicController.php:376]: Passing `$lead` completely bypasses the logic in [PageModel](https://github.com/mautic/mautic/blob/staging/app/bundles/PageBundle/Model/PageModel.php#L462-L479). This alters current application behaviour. E.g. `ct` query parameter is not taken into account, etc.

File: app/bundles/PageBundle/Controller/PublicController.php [hunk 3]
```diff
@@ -402,6 +404,10 @@ public function trackingAction()
         /** @var TrackingHelper $trackingHelper */
         $trackingHelper = $this->get('mautic.page.helper.tracking');
         $sessionValue   = $trackingHelper->getSession(true);
+        
+        /** @var \Mautic\PageBundle\Model\PageModel $model */
+        $model = $this->getModel('page');
+        $model->hitPage(null, $this->request, '200', $lead);
```
Comment (fedys) [inline app/bundles/PageBundle/Controller/PublicController.php:410]: Passing `$lead` completely bypasses the logic in [PageModel](https://github.com/mautic/mautic/blob/staging/app/bundles/PageBundle/Model/PageModel.php#L462-L479). This alters current application behaviour. E.g. `ct` query parameter is not taken into account, etc.

**Review by fedys — CHANGES_REQUESTED** [review]

**Comment by RCheesley** [comment]
@xnohat please can you review the feedback?

**Comment by mabumusa1** [comment]
@xnohat this is a valuable PR, we would love to get your contribution, I hope you have some time to review the comments and publish the changes. I would love to test it again

**Comment by npracht** [comment]
@xnohat do you plan to continue working on this PR or i should close it ?
Thanks a lot for your help :)

**Pull request closed without merging**